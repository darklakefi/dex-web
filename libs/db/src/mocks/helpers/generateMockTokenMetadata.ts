import { randAlphaNumeric, randCompanyName, randUrl } from "@ngneat/falso";
import type { NewTokenMetadata } from "../../schemas/types";
import { generateMockSolanaAddress } from "./generateMockSolanaAddress";

export function generateMockTokenMetadata(count: number): NewTokenMetadata[] {
  const decimalsOptions = [6, 8, 9];
  return Array.from({ length: count }, () => {
    const randomIndex = Math.floor(Math.random() * decimalsOptions.length);
    const decimals =
      typeof decimalsOptions[randomIndex] === "number"
        ? decimalsOptions[randomIndex]
        : 6;
    return {
      decimals,
      name: `${randCompanyName()} Token`,
      symbol: Array.from({ length: Math.floor(Math.random() * 4) + 3 }, () =>
        randAlphaNumeric(),
      )
        .join("")
        .toUpperCase(),
      token_address: generateMockSolanaAddress(),
      uri: randUrl(),
    };
  });
}
