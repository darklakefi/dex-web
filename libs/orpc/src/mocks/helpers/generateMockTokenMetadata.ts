import { randCurrencySymbol, randNumber, randUuid, seed } from "@ngneat/falso";
import { type DAS, Interface, OwnershipModel } from "helius-sdk";
import { FIXED_SEED } from "./constants";

export function generateMockTokenMetadata(): DAS.GetAssetResponse {
  seed(FIXED_SEED);
  return {
    burnt: false,
    id: randUuid(),
    interface: Interface.V1NFT,
    mutable: true,
    ownership: {
      delegate: undefined,
      delegated: false,
      frozen: true,
      owner: randUuid(),
      ownership_model: OwnershipModel.SINGLE,
    },
    token_info: {
      decimals: randNumber({ max: 9, min: 0 }),
      symbol: randCurrencySymbol(),
    },
  };
}
