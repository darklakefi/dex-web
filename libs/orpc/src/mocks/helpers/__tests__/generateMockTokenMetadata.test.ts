import { generateMockTokenMetadata } from "../generateMockTokenMetadata";

describe("generateMockTokenMetadata", () => {
  it("should generate a list of token accounts", () => {
    const response = generateMockTokenMetadata();

    expect(response).toMatchSnapshot();
    expect(response.token_info).toBeDefined();
    expect(response.token_info?.decimals).toBeDefined();
    expect(response.token_info?.symbol).toBeDefined();
  });
});
