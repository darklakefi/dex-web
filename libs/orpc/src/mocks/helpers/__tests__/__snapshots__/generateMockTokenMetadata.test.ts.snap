// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateMockTokenMetadata > should generate a list of token accounts 1`] = `
{
  "burnt": false,
  "id": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
  "interface": "V1_NFT",
  "mutable": true,
  "ownership": {
    "delegate": undefined,
    "delegated": false,
    "frozen": true,
    "owner": "e7281d9e-834d-4e58-a56c-106dad7426e4",
    "ownership_model": "single",
  },
  "token_info": {
    "decimals": 5,
    "symbol": "â‚¡",
  },
}
`;
